transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionMappers.java

```diff
@@ -0,0 +1,49 @@
+package com.example.auction.transaction.impl;
+
+import com.example.auction.transaction.api.DeliveryInfo;
+import com.example.auction.transaction.api.TransactionInfo;
+
+import java.util.Optional;
+
+public class TransactionMappers {
+
+    public static Optional<DeliveryInfo> toApi(Optional<DeliveryData> data) {
+        if(data.isPresent()) {
+            return Optional.of(new DeliveryInfo(
+                    data.get().getAddressLine1(),
+                    data.get().getAddressLine2(),
+                    data.get().getCity(),
+                    data.get().getState(),
+                    data.get().getPostalCode(),
+                    data.get().getCountry()
+            ));
+        }
+        else
+            return Optional.empty();
+    }
```

Comment:
This is a very common antipattern of `Optional`. A `if(opt.isPresent) {...} else {Optional.empty}` is equivalent to `Optional.map()`. The code can be rewritten as:

```java
        return data.map(deliveryData -> new DeliveryInfo(
                deliveryData.getAddressLine1(),
                deliveryData.getAddressLine2(),
                deliveryData.getCity(),
                deliveryData.getState(),
                deliveryData.getPostalCode(),
                deliveryData.getCountry()
        ));
```

NOTE: IntelliJ made the suggestion and the rewrite for me. IntelliJ detected the antipattern and marked the code in brown, than using `Alt-return` made the fix.